datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/DatastreamResources.java

```diff
@@ -97,33 +99,57 @@
   private static final String CREATE_CALL_LATENCY_MS_STRING = "createCallLatencyMs";
   private static final String DELETE_CALL_LATENCY_MS_STRING = "deleteCallLatencyMs";
 
+  // To support retries on the request timeouts
+  public static final String CONFIG_STOP_TRANSITION_TIMEOUT_MS = "stopTransitionTimeoutMs";
+  public static final String CONFIG_STOP_TRANSITION_RETRY_PERIOD_MS = "stopTransitionRetryPeriodMs";
+  private static final Duration STOP_TRANSITION_TIMEOUT_MS_DEFAULT = Duration.ofMillis(60000);
+  private static final Duration STOP_TRANSITION_RETRY_PERIOD_MS_DEFAULT = Duration.ofMillis(1000);
+
   private final DatastreamStore _store;
   private final Coordinator _coordinator;
   private final ErrorLogger _errorLogger;
 
   private final DynamicMetricsManager _dynamicMetricsManager;
 
+  private Duration _stopTransitionTimeoutMs = STOP_TRANSITION_TIMEOUT_MS_DEFAULT;
+  private Duration _stopTransitionRetryPeriodMs = STOP_TRANSITION_RETRY_PERIOD_MS_DEFAULT;
+
   /**
    * Constructor for DatastreamResources
    * @param datastreamServer the datastream server
    */
   public DatastreamResources(DatastreamServer datastreamServer) {
-    this(datastreamServer.getDatastreamStore(), datastreamServer.getCoordinator());
+    this(datastreamServer.getDatastreamStore(), datastreamServer.getCoordinator(), datastreamServer.getProperties());
   }
 
   /**
    * Constructor for DatastreamResources
    * @param store the datastream store
    * @param coordinator the server coordinator
    */
-  public DatastreamResources(DatastreamStore store, Coordinator coordinator) {
+  public DatastreamResources(DatastreamStore store, Coordinator coordinator, Properties properties) {
```

Comment:
nit: You could have created another construction marked as @VisibleForTesting with store and coordinator fields and call DatastreamResources(store, coordinator, null); This would have avoided most of the test changes required for this new definition.